tests/test_colormap.py

```diff
@@ -55,3 +56,28 @@ def test_step_object():
     cm.step.PuBu_06.to_linear()
     cm.step.YlGn_06.scale(3, 12)
     cm.step._repr_html_()
+
+@pytest.mark.parametrize("max_labels,expected", [
+    (10, [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0]),
+    (5, [0.0, '', 2.0, '', 4.0, '', 6.0, '', 8.0, '']),
+    (3, [0.0, '', '', '', 4.0, '', '', '', 8.0, '', '', '']),
+])
+def test_max_labels_linear(max_labels, expected):
+    colorbar = cm.LinearColormap(['red'] * 10, vmin=0, vmax=9, max_labels=max_labels)
+    try:
+        colorbar.render()
+    except AssertionError:
+        assert colorbar.tick_labels == expected
```

Comment:
nit: I was a bit confused by this at first. I was thinking, what does that `assert` have to do with that `AssertionError`? But I understand now that you are catching the error created because the colorbar doesn't have a Figure as parent.

In this case I would split these statements, so it's clear that the `assert` here doesn't relate to the `AssertionError` you are catching.

```suggestion
    try:
        colorbar.render()
    except AssertionError:
        pass
    assert colorbar.tick_labels == expected
```

This is a bit of a detail, so I don't mind keeping it as is.

Reply:
Sure. I'll also add a comment with a short explanation for other reading the code.